src/main/java/io/vertx/ext/mail/impl/MailClientImpl.java

```diff
@@ -115,15 +115,15 @@ public MailClient sendMail(MailMessage message, Handler<AsyncResult<MailResult>>
   }
 
   private void getConnection(MailMessage message, Handler<AsyncResult<MailResult>> resultHandler, Context context) {
-    connectionPool.getConnection(hostname, result -> {
+    context.runOnContext(v -> connectionPool.getConnection(hostname, context, result -> {
```

Comment:
I don't think runOnContext is needed here, it can be the current thread.

Reply:
Yes, it makes sense, I updated it accordingly. thanks :)